### ISSUE TITLE
performance issue privateGPT 

### ISSUE THREAD

**Issue by mehrdad2000** [imartinez/privateGPT#372]
Hi
I've consider 40 core cpu/ 40GB mem/ssd disk for privateGPT but still slowing when it will answer!

Any idea or trick to improve performance?

FYI: set n_threads=40 in this file privateGPT.py to use all cpu cores.

Thanks

**Comment by 2blackbar** [comment]
gpu support has been just cracked so it should be soon merged with main branch

**Comment by mehrdad2000** [comment]
@2blackbar how it work? If it merge still required gpu or it will work with cpu with better performance?

**Comment by darrinh** [comment]
GPU support will be awesome for inference, the ingestion is super fast as it is on CPU. Looking forward to trying it.

thanks
Darrin

**Comment by mehrdad2000** [comment]
@darrinh my issue about slow answer not slow ingest!

**Comment by darrinh** [comment]
Yes, I have the same issue, was saying that ingest works fine as it is on CPU.

**Comment by mehrdad2000** [comment]
@2blackbar any update? Does it affect speed of answer or just improve ingest time?
